plutus-core/plutus-core/src/PlutusCore/Name/UniqueSet.hs

```diff
@@ -0,0 +1,91 @@
+module PlutusCore.Name.UniqueSet (
+  UniqueSet (..),
+  insertByUnique,
+  insertByName,
+  singletonName,
+  fromFoldable,
+  fromUniques,
+  fromNames,
+  memberByUnique,
+  memberByName,
+  notMemberByName,
+  (\\),
+  union,
+  setOfByUnique,
+  setOfByName,
+) where
+
+import Control.Lens (Getting, view)
+import Control.Lens.Getter (views)
+import Data.Coerce (Coercible, coerce)
+import Data.Foldable (foldl')
+import Data.IntSet qualified as IS
+import Data.IntSet.Lens qualified as IS
+import PlutusCore.Name.Unique (HasUnique (..), Unique (Unique))
+
+-- | A set containing uniques.
```

Comment:
Why is this good? I have honestly forgotten, and it's not in the PR description, or more importantly, in the doc for this type. How will we remember when to use it or not? What were the considerations that led to us defining this? I'm sure you know - write it down!

Reply:
Same applies to `UniqueMap`